crates/core/tedge_agent/src/tedge_to_te_converter/tests.rs

```diff
@@ -327,12 +329,12 @@ async fn spawn_tedge_to_te_converter(
 
     // Tedge to Te converter
     let mut tedge_converter_actor =
-        ConvertingActor::builder("TedgetoTeConverter", tedge_to_te_converter, subscriptions);
+        ConvertingActor::builder("TedgetoTeConverter", tedge_to_te_converter);
 
-    let mqtt_box = SimpleMessageBoxBuilder::new("MQTT", 5)
-        .with_connection(&mut tedge_converter_actor)
-        .build()
-        .with_timeout(Duration::from_millis(100));
+    let mut mqtt_box = SimpleMessageBoxBuilder::new("MQTT", 5);
+    mqtt_box.register_peer(subscriptions, tedge_converter_actor.get_input_sender());
+    tedge_converter_actor.register_peer(NoConfig, mqtt_box.get_sender());
```

Comment:
Yeah, the `set_connection()` method is convenient. However, we need to avoid the previous issues conflating the two types of Config and too specialized on a mutual-connection (MQTT being used both as a source and a sink) ignoring the pipeline case (see for instance the batching actor used for the collectd mapper).

This can be discussed along renaming of the Sink and Source methods.